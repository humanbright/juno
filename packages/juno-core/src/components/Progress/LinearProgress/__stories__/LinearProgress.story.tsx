import React, { ComponentProps } from 'react';

import { RcLinearProgress } from '@ringcentral/juno';
import {
  notControlInDocTable,
  notShowInDocTable,
  paletteChoice,
  sortInDocTable,
  switchToControlKnobs,
  Title,
} from '@ringcentral/juno-storybook';
import { Meta, Story } from '@storybook/react';

export default {
  title: 'ðŸš€ Cleanup Components/Progress/LinearProgress',
  component: RcLinearProgress,
  argTypes: {
    color: {
      options: paletteChoice,
      control: {
        type: 'select',
      },
    },
    ...sortInDocTable<keyof LinearProgressProps>([
      'variant',
      'value',
      'valueBuffer',
    ]),
    ...notControlInDocTable<keyof LinearProgressProps>([]),
    ...notShowInDocTable<keyof LinearProgressProps>([]),
  },
} as Meta;

type LinearProgressProps = ComponentProps<typeof RcLinearProgress>;

export const LinearProgress: Story<LinearProgressProps> = ({ ...args }) => {
  switchToControlKnobs();
  return <RcLinearProgress {...args} />;
};

LinearProgress.storyName = 'LinearProgress';

LinearProgress.args = {};

LinearProgress.argTypes = {
  ...notControlInDocTable<keyof LinearProgressProps>([]),
};
LinearProgress.parameters = {
  tags: [
    {
      name: 'Mui',
      href: 'https://material-ui.com/components/progress/',
    },
    {
      name: 'Accessibility',
      value: '100%',
    },
  ],
};

export const LinearProgressExample: Story<LinearProgressProps> = ({
  color,
}) => {
  switchToControlKnobs();
  return (
    <div>
      <Title>Default</Title>
      <RcLinearProgress color={color} />
      <br />
      <br />
      <Title>Determinate with value</Title>
      <RcLinearProgress color={color} variant="determinate" value={25} />
      <br />
      <br />
      <Title>Query progress</Title>
      <RcLinearProgress color={color} variant="query" value={25} />
      <br />
      <br />
      <Title>Buffer progress</Title>
      <RcLinearProgress
        color={color}
        variant="buffer"
        value={25}
        valueBuffer={50}
      />
    </div>
  );
};

LinearProgressExample.storyName = 'LinearProgress Examples';

LinearProgressExample.args = {
  color: 'danger.f02',
};
