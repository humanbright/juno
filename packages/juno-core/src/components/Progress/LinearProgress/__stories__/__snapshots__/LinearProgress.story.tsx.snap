// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Progress/LinearProgress LinearProgress 1`] = `
.c0:not(.RcLinearProgress-buffer),
.c0 .RcLinearProgress-colorPrimary {
  background-color: rgb(160,200,223);
}

.c0 .RcLinearProgress-barColorPrimary {
  background-color: #066FAC;
}

.c0 .RcLinearProgress-dashedColorPrimary {
  background-image: radial-gradient( rgb(160,200,223) 0%, rgb(160,200,223) 16%, transparent 42% );
}

<div
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary RcLinearProgress-colorPrimary c0 MuiLinearProgress-indeterminate"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary RcLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary RcLinearProgress-barColorPrimary"
    style={Object {}}
  />
</div>
`;

exports[`Storyshots ðŸš€ Cleanup Components/Progress/LinearProgress LinearProgress Examples 1`] = `
.c1:not(.RcLinearProgress-buffer),
.c1 .RcLinearProgress-colorPrimary {
  background-color: rgb(230,179,177);
}

.c1 .RcLinearProgress-barColorPrimary {
  background-color: #BE3933;
}

.c1 .RcLinearProgress-dashedColorPrimary {
  background-image: radial-gradient( rgb(230,179,177) 0%, rgb(230,179,177) 16%, transparent 42% );
}

.c0 {
  font-size: 1.5rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 32px;
  color: #121212;
}

<div>
  <h5
    className="MuiTypography-root c0 MuiTypography-paragraph MuiTypography-displayBlock"
    data-color="neutral.f06"
    data-variant="headline1"
  >
    Default
  </h5>
  <div
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary RcLinearProgress-colorPrimary c1 MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary RcLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary RcLinearProgress-barColorPrimary"
      style={Object {}}
    />
  </div>
  <br />
  <br />
  <h5
    className="MuiTypography-root c0 MuiTypography-paragraph MuiTypography-displayBlock"
    data-color="neutral.f06"
    data-variant="headline1"
  >
    Determinate with value
  </h5>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary RcLinearProgress-colorPrimary c1 MuiLinearProgress-determinate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary RcLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
      style={
        Object {
          "transform": "translateX(-75%)",
        }
      }
    />
  </div>
  <br />
  <br />
  <h5
    className="MuiTypography-root c0 MuiTypography-paragraph MuiTypography-displayBlock"
    data-color="neutral.f06"
    data-variant="headline1"
  >
    Query progress
  </h5>
  <div
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary RcLinearProgress-colorPrimary c1 MuiLinearProgress-query"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary RcLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary RcLinearProgress-barColorPrimary"
      style={Object {}}
    />
  </div>
  <br />
  <br />
  <h5
    className="MuiTypography-root c0 MuiTypography-paragraph MuiTypography-displayBlock"
    data-color="neutral.f06"
    data-variant="headline1"
  >
    Buffer progress
  </h5>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={25}
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary RcLinearProgress-colorPrimary c1 MuiLinearProgress-buffer RcLinearProgress-buffer"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-dashed MuiLinearProgress-dashedColorPrimary RcLinearProgress-dashedColorPrimary"
    />
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary RcLinearProgress-barColorPrimary MuiLinearProgress-bar1Buffer"
      style={
        Object {
          "transform": "translateX(-75%)",
        }
      }
    />
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-colorPrimary RcLinearProgress-colorPrimary MuiLinearProgress-bar2Buffer"
      style={
        Object {
          "transform": "translateX(-50%)",
        }
      }
    />
  </div>
</div>
`;
